scalyr_agent/agent_main.py

```diff
@@ -689,6 +700,40 @@ def __handle_terminate(self):
             log.info("Received signal to shutdown, attempt to shutdown cleanly.")
             self.__run_state.stop()
 
+    def _get_system_and_agent_stats(self):  # type: () -> Optional[Dict]
+        """
+        Return current machine's and agent process' stats.
+        :return: Dict with stats.
+        """
+        if psutil is None:
+            return None
+
+        pidfile = os.path.join(self.__config.agent_log_path, "agent.pid")
+
+        if not os.path.exists(pidfile):
+            raise Exception("Pid file does not exist.")
+
+        with open(pidfile, "r") as fp:
+            pidfile_content = fp.read().strip()
+
+        process = psutil.Process(
+            pid=int(pidfile_content)
+        )
+
+        return {
```

Comment:
Per slack discussion the other day, I'm slightly worried that this logging will not be sufficient to track down the root cause.

I propose adding this additional logging:

* Use psutil or similar to run command similar to "ps aux | grep scalyr-agent-2" and log the output. This may help us catch an issue if there is a stuck process from a previous run or similar.
* This information should be logged multiple times - before executing status, during (while sleeping and waiting) and at the end. That's important because it's likely a race condition or similar and printing it just at the end likely won't be sufficient.